cmake/FindGMP.cmake

```diff
@@ -0,0 +1,21 @@
+# Try to find the GNU Multiple Precision Arithmetic Library (GMP)
```

Comment:
I'd suggest using the [FindGMP.cmake](https://github.com/libigl/libigl/tree/jdumas/cmake-refactor/cmake/find) from my libigl, accompanied by the following [gmp.cmake](https://github.com/libigl/libigl/blob/fec2d43ff403e59a4f0046584642df17441d1e58/cmake/recipes/external/gmp.cmake#L1) recipe file (+ the `gmp_mpfr.cmake` to download the .dll on Windows).